commercial/app/controllers/commercial/JobsController.scala

```diff
@@ -22,14 +26,24 @@ class JobsController(jobsAgent: JobsAgent) extends Controller with implicits.Req
     JobSector("media", "Media")
   )
 
+  private def jobSample(specificIds: Seq[String], segment: Segment): Seq[Job] =
+    (jobsAgent.specificJobs(specificIds) ++ jobsAgent.jobsTargetedAt(segment)).distinct.take(2)
+
   def renderJobs = Action { implicit request =>
-    (jobsAgent.specificJobs(specificIds) ++ jobsAgent.jobsTargetedAt(segment)).distinct match {
+    jobSample(specificIds, segment) match {
       case Nil => NoCache(jsonFormat.nilResult.result)
```

Comment:
I know you have not modified it but why are we not caching the empty result (even for a very short period of time)?
Not caching it at all means that if we return no jobs, all requests made to this endpoint will hit our origin as long as we keep returning an empty results.